### DISCUSSION TITLE
questions

### DISCUSSION THREAD

**Discussion by Mariaab1** [NHAS/wag#135] [Q&A]
Hello it's me again 😅,

I have a series of questions:

1. Is it necessary to always use a cluster? I removed the "clustering" object from my config.json, but it still asks for it:

`root@vpn-clonada:/opt/wag_8_1# ./wag start
2024/10/15 11:54:02 no listen url was specified for the etcd cluster TLS manager, new wag servers will not be able to join`

2. Why don’t the assigned IPs follow a sequential order?

![image](https://github.com/user-attachments/assets/73582f1a-cfe4-4323-98bb-3e90869dcb1e)

Is there any way to change this?

3. Why does WireGuard set the iptables FORWARD chain policy to DROP every time the service is started?

![image](https://github.com/user-attachments/assets/664aad45-30f9-4d0d-9ebd-2197976b9b20)

4. When I create a user, a configuration file is generated, but it only allows a specific IP:

![image](https://github.com/user-attachments/assets/d2a741c1-8c9c-4095-9f05-fa684a89d07b)

However, if I later add rules granting access to a network segment:

![image](https://github.com/user-attachments/assets/12dd078f-fbca-4366-a779-72104202d109)

The changes don’t take effect:

![image](https://github.com/user-attachments/assets/74a55514-2a18-4968-9341-23c570c34a62)

 However, if I manually modify the WireGuard configuration file, everything works as expected:

![image](https://github.com/user-attachments/assets/cb933348-93a6-4715-b777-c773bcc66629)
![image](https://github.com/user-attachments/assets/c39ce350-a9d5-4994-9560-d18a7ad54f03)

 
Does this mean that any changes I make must be manually applied to each user's WireGuard configuration file in order to take effect?
 
Thank you!🐧

**Comment by NHAS** [comment]
1. Wag uses ETCd to manage its data, you can have a single node cluster. But its always a cluster
2. The IPs dont get assigned sequentially, as its easier to find "gaps" where devices use to have an associated IP but have since been deleted, its a pretty fast way to select ip addresses over all. This cant be changed.
3. Wag wants to make sure that random traffic isnt forwarded and that only wireguard interface traffic is being forwarded, you can make this change yourself, but this is how wag functions at the moment. 
4. Wireguard plain has no mechanism to update configuration files. You have to update your wireguard config yourself, wag gives you tools to be able to do this automatically, but cant do it just via wireguard. 

Checkout the `/routes/` path in the vpn tunnel.

**Comment by Mariaab1** [comment]
Okay, thank you very much. Sorry, I hadn’t noticed that it could be opened as a "discussion".

I also wanted to mention that when I generate a token for a new user and add them to an existing group, once I execute the curl command to register them, they are registered but the group I indicated is ignored:

![image](https://github.com/user-attachments/assets/c28ec13b-0755-4a21-ad85-1ab7c92b40bd)
![image](https://github.com/user-attachments/assets/e8a70486-7856-41f0-90d3-d1529416a734)
![image](https://github.com/user-attachments/assets/cfd9ae76-5360-45a4-88c5-8b873fae4fb6)


Additionally, I wanted to let you know that maybe you might want  to add a trim function when adding users to groups, as it does not recognize them if there’s a blank space

![image](https://github.com/user-attachments/assets/90efb572-1efa-4661-81a5-6e4ceae8ec03)
![image](https://github.com/user-attachments/assets/84b2faeb-252a-40dd-a13a-e781c6906273)

However, if I remove the trailing space...🎉

![image](https://github.com/user-attachments/assets/72fb391a-10de-4f97-8b6c-7a45a7a70bca)
![image](https://github.com/user-attachments/assets/bab48d3d-1934-47a4-88f8-4b8a2f9c2701)

**Comment by NHAS** [comment]
Ah cool, I'll fix the whitespace issue. Just working on some other bigger changes that are taking precedence, that group issue is a known one but its more of an intended design weirdly enough. 

Its for when you have oidc members that dont initially have groups assigned to them before log in, and sometimes you want to give them access to say the SSO provider as well

**Comment by Mariaab1** [comment]
Ah, yes, I understand. Well, if I spot anything else, I'll let you know 😌

I wanted to ask you one last thing (for now 😅), I have this error:

![image](https://github.com/user-attachments/assets/566edd9c-6eb0-4210-a3dd-010cde8d2942)


But the thing is that in the config JSON, it's clearly defined:

![image](https://github.com/user-attachments/assets/8d013ffe-ab9d-4ffa-bcb1-e9e7d60460d5)


 The problem is that when the user tries to connect, they use an authentication method that isn't set:
 
 `2024/10/21 09:09:46 prueba 172.16.101.11 Invalid MFA type requested:  webauthn` 
 
 And when I try to configure it from the GUI, the list of options doesn't appear 😔:
 
 
![image](https://github.com/user-attachments/assets/d629266c-f2c1-4789-889e-7bed288618bb)

(It doesn't look exactly like that because I've reinstalled it from scratch, and now it works. So the screenshot is recreated, but basically, the dropdown didn't expand and was empty)

thank you!

**Reply by NHAS** [reply]
I can understand the confusion, this is because we're moving away from using the JSON config file and more towards the web UI. 

You can see that in your list of "Enabled MFA Methods" there is no enabled method. Try enabling totp, if that is still broken then its a bug :)

**Reply by Mariaab1** [reply]
Hello!

Then I would say yes (about the bug).I’ve tried again and this is the actual error. Even though I have "DefaultMethod" set in config.json, this happens in the GUI.

![image](https://github.com/user-attachments/assets/41337dd4-f9a4-47e5-a03a-d3effc9a8b94)
![image](https://github.com/user-attachments/assets/9dbfa36b-50aa-422c-9286-5448a0e619ee)

I was also wondering if, in case I need to migrate it, simply copying the "default.wag-node.etcd" directory would be enough. I've done some tests, and so far it seems to work.

Thank you very much!

**Reply by NHAS** [reply]
Sweet. That's a good bug find. At the moment I'm working on some big changes that remove ebpf from wag. So I'll go and fix this as an issue after that work is largely completed
